pkg/bmc/ipmi.go

```diff
@@ -9,11 +9,24 @@ func init() {
 	RegisterFactory("libvirt", newIPMIAccessDetails, []string{})
 }
 
+func getPrivLevel(rawquery string) string {
+	privlevel := "ADMINISTRATOR"
+	q, ok := url.ParseQuery(rawquery)
+	if ok != nil {
+		return privlevel
+	}
+	if val, ok := q["privlevel"]; ok {
+		return val[0]
+	}
+	return privlevel
+}
+
 func newIPMIAccessDetails(parsedURL *url.URL, disableCertificateVerification bool) (AccessDetails, error) {
 	return &ipmiAccessDetails{
 		bmcType:                        parsedURL.Scheme,
 		portNum:                        parsedURL.Port(),
 		hostname:                       parsedURL.Hostname(),
+		privlevel:                      getPrivLevel(parsedURL.RawQuery),
```

Comment:
Can we call this PrivilegeLevel instead? Prefer not to use abbreviations to avoid confusion

Reply:
will do